plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java

```diff
@@ -70,5 +79,12 @@ protected void handleMLTaskComplete(MLTask mlTask) {
         }
     }
 
-    public abstract void run(Request request, TransportService transportService, ActionListener<Response> listener);
+    public void run(Request request, TransportService transportService, ActionListener<Response> listener) {
+        if (mlCircuitBreakerService.isOpen()) {
+            throw new MLLimitExceededException("Circuit breaker is open");
+        }
+        executeTask(request, transportService, listener);
+    }
+
+    public abstract void executeTask(Request request, TransportService transportService, ActionListener<Response> listener);
```

Comment:
This circuit breaker should be only applied to train/execute tasks that require many resources right? I assume the other API tasks wouldn't need it since they only run a light query to the cluster? or maybe the cluster has that circuit breaker implemented already.

Reply:
Correct. The train/predict task will consume more resource, so we need to check circuit breaker.